src/monitor/group_state_machine.c

```diff
@@ -1256,8 +1263,45 @@ ProceedGroupStateForMSFailover(AutoFailoverNode *activeNode,
 	 * So all the expected candidates did report their LSN, no node is missing.
 	 * Let's see about selecting a candidate for failover now, when we do have
 	 * candidates.
+	 *
+	 * To start the selection process, we require at least number_sync_standbys
+	 * nodes to have reported their LSN and be currently healthy, otherwise we
+	 * won't be able to maintain our guarantees: we would end-up with a node in
+	 * WAIT_PRIMARY state with all the writes blocked for lack of standby
+	 * nodes.
 	 */
-	if (candidateList.candidateCount > 0)
+	int minCandidates = formation->number_sync_standbys + 1;
+
+	/* no candidates is a hard pass */
+	if (candidateList.candidateCount == 0)
+	{
+		return false;
+	}
+
+	/* not enough candidates to promote and then accept writes, pass */
+	else if (candidateList.quorumCandidateCount < minCandidates)
+	{
+		char message[BUFSIZE] = { 0 };
+
+		LogAndNotifyMessage(
+			message, BUFSIZE,
+			"Failover still in progress with %d candidates having reported "
```

Comment:
Let's make this error message a bit more explicit to avoid confusion:

```suggestion
			"Failover still in progress with %d candidates that participate in the quorum having reported "
```

Reply:
Thanks, added.